app/auth/__init__.py

```diff
@@ -1,68 +1,132 @@
 import traceback
 from functools import wraps
-from flask import jsonify, request
+from flask import jsonify, request, abort
 from flask_jwt_simple import JWTManager, create_jwt, get_jwt_identity, jwt_required
 from jwt.exceptions import PyJWTError
 from flask_jwt_simple.exceptions import InvalidHeaderError, NoAuthorizationError
-from auth.ldap import LdapAuthenticator
+
 from app import logger
-from utils.env_loader import config
+from auth.ldap import LdapAuthenticator
+from auth.exceptions import ApplicationUserRoleException
+from auth.authenticator import EmptyAuthenticator
+from models import db
+from models.user import User
+from models.application import Application
+from models.application_user_role import ApplicationUserRole, Role
 
 
 class Auth(object):
-    enabled = False
-
-    def __init__(self, app=None):
-        if app is not None:
-            self.init_app(app)
+    __enabled = False
 
     @staticmethod
     def auth_required(fn):
+        def check_role(user_id, application_id, method):
```

Comment:
Is objective of this method same as [here](https://github.com/rekcurd/drucker-dashboard/pull/49/files#diff-c6aa010351cf936dcad1354eea9e2b62R36)? If so, I want to unify them.

Reply:
No, objectives are different.
This function is used to judge whether or not to operate the application. only editor or the higher users can change application.
check_owner_role is used to judge whether or not to change authorized users. Only owner users can change them.

Reply:
I see. But almost same.
l24 - l33 will be a method and we can use it in this method and `check_owner_role` method [here](https://github.com/rekcurd/drucker-dashboard/pull/49/files#diff-c6aa010351cf936dcad1354eea9e2b62R41).

Reply:
fixed.
https://github.com/rekcurd/drucker-dashboard/pull/49/commits/03ab0bf05a9a0192e62f85a6be0fd516a548ac04